cmd/libsnap-confine-private/tool.c

```diff
@@ -167,7 +167,10 @@ static int sc_open_snapd_tool(const char *tool_name)
 		die("cannot open path %s", dir_name);
 	}
 	int tool_fd = -1;
-	tool_fd = openat(dir_fd, tool_name, O_PATH | O_NOFOLLOW | O_CLOEXEC);
+	/* Use O_CLOEXEC but not O_PATH. This allows us to determine if the file is
+	 * a binary or a script just prior to passing it to fexecve. This way we
+	 * can still undo the use of O_CLOEXEC which is a safe default. */
+	tool_fd = openat(dir_fd, tool_name, O_RDONLY | O_NOFOLLOW | O_CLOEXEC);
```

Comment:
I feel uneasy that we have to modify the code to accommodate for a test in this way. Maybe we can modify the test a bit, something like: http://paste.ubuntu.com/p/Z2k4z2rdCh/  ? I.e. do not use shell for the test but just build a tiny go binary and just run the test on systems that have go (or alternatively install go as a snap where its needed and skip the test on ubuntu-core systems only) or alternatively put a binary that sleeps into a snap and copy that in place if we want the test to run everywhere (but given that is has nothing core or classic specific it seems to be ok to run it on a subset of the systems only?).